python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py

```diff
@@ -73,6 +74,7 @@ def __init__(
         extensions_directory=None,
         check_consistency=False,
         device=None,
+        properties_to_store: Optional[List[str]] = None,
```

Comment:
```suggestion
        extra_metatensor_outputs: Optional[Dict[str, ModelOutput]] = None,
```

I would rename this and handle it a bit differently from "normal" properties to make it clearer that this mechanism is only intended for things which are *not* already known ASE properties. 

IMO the data should also be stored as TensorMap in `calc.results["extra_metatensor_outputs"]["..."]`, again to make it clear we are operating outside of the ASE interface.

Reply:
I agree that it would be good in principle but it breaks the ASE caching mechanism and it turns it de facto into a different calculator with no re-use (which is pointless because we already have the function to call the model separately)

Reply:
> but it breaks the ASE caching mechanism

How? Especially if we only allow things ASE has no knowledge about in `extra_metatensor_outputs`?

> it turns it de facto into a different calculator with no re-use

No? There will be a single model call to compute both "energy", "forces" and e.g. "last_layer_features". Any weight reuse is then up to the model.